src/modules/class/application/create-class.dto.ts

```diff
@@ -0,0 +1,35 @@
+
+import { ApiProperty } from "@nestjs/swagger";
+import { IsArray, IsNotEmpty, IsNumber, IsOptional, IsString } from "class-validator"
+
+export class CreateClassDTO {
+    @ApiProperty({ example: "Turma de violão", description: "Nome da turma" })
+    @IsString()
+    @IsNotEmpty()
+    name: string;
+
+    @ApiProperty({ example: 1, description: "ID da atividade" })
+    @IsNumber()
+    @IsNotEmpty()
+    activityId: number;
+
+    @ApiProperty({ example: 1, description: "ID do nível" })
+    @IsNumber()
+    @IsNotEmpty()
+    levelId: number;
+
+    @ApiProperty({ example: "ATIVA", description: "Estado da turma" })
+    @IsString()
+    @IsNotEmpty()
+    state: string;
+
+    @ApiProperty({ example: [1, 2, 3], description: "Lista de IDs de alunos", required: false })
+    @IsArray()
+    @IsOptional()
+    studentsIds: number[] |  null
```

Comment:
```suggestion
    @ApiProperty({ example: [1, 2, 3], description: "Lista de IDs de alunos", required: false })
    @IsArray()
    @IsOptional()
    @IsNumber({}, { each: true })
    @Type(() => Number)
    studentsIds: number[] |  null
```
A verificação de se os dados são numéricos e a transformação deles caso sejam devem ser feita no próprio DTO.